.circleci/config.yml

```diff
@@ -92,385 +108,894 @@ commands:
+                exit 0
+              elif [ "$(kubectl get pods | grep -c 'Completed')" -eq "1" ]; then
+                echo "Pod for << parameters.image_name >> has completed"
+                exit 0
+              elif [ "$(kubectl get pods | grep -c 'Error')" -eq "1" ] || [ "$(kubectl get pods | grep -c 'CrashLoopBackOff')" -eq "1" ]; then
+                echo "Pod for << parameters.image_name >> is in an error state"
+                kubectl logs << parameters.pod_name >>
+                exit 1
+              fi
+              sleep << parameters.sleep_time >>
+            done
+          no_output_timeout: "5m"
+      - run:
+          # todo: make getting the exit_code more robust: add check for "commands succeeded"
+          name: "Wait for tests to finish, get pod logs, and parse exit code"
+          command: |
+            sleep 30
+            
+            while true; do
+              kubectl get pods
+              if [ "$(kubectl get pods | grep -c 'Running')" -eq "1" ]; then
+                echo "Pod for << parameters.image_name >> is running"
+              elif [ "$(kubectl get pods | grep -c 'Completed')" -eq "1" ]; then
+                echo "Pod for << parameters.image_name >> has completed"
+                break
+              elif [ "$(kubectl get pods | grep -c 'Error')" -eq "1" ] || [ "$(kubectl get pods | grep -c 'CrashLoopBackOff')" -eq "1" ]; then
+                echo "Pod for << parameters.image_name >> is in an error state"
+                kubectl logs << parameters.pod_name >>
+                exit 1
+              fi
+              sleep << parameters.sleep_time >>
+            done
+            
+            kubectl logs << parameters.pod_name >>
+            logs=`kubectl logs << parameters.pod_name >>`
+            exit_code=`echo $(echo $logs | grep -c "commands failed")`
+            echo "Pod for << parameters.image_name >> exited with code ${exit_code}"
+            exit ${exit_code}
+          no_output_timeout: "20m"
+      - run:
+          name: "Delete the pod"
+          when: always
+          command: |
+            kubectl get pods
+            kubectl delete -f << parameters.path >>/<< parameters.pod_config_name >>
+      # conditionally post a notification to slack if the job failed
+      - when:
+          condition: << parameters.notify_on_failure >>
+          steps:
+            - slack/notify:
+                event: fail
+                template: basic_fail_1
+                # taken from slack-secrets context
+                channel: $SLACK_SDK_NIGHTLY_CI_CHANNEL
+
 workflows:
+  nightly:
+    when:
+      and:
+        - equal:
+            - scheduled_pipeline
+            - << pipeline.trigger_source >>
+        - equal:
+            - "nightly"
+            - << pipeline.schedule.name >>
+    jobs:
+      - slack_notify:
+          name: "slack-notify-on-start"
+          context: slack-secrets
+          message: ":runner: *Nightly run started!*"
```

Comment:
should we also add  a link to the run that started?

Reply:
too much noise perhaps? there will be now be success messages, but I'll ad a todo anyways

Reply:
ah no that's fine... one thing I was thinking... don't know if it is supported or how hard to do it... 
that the success/failure message be a thread after the initial run started message ... what do you think?

Reply:
yea, that'd be awesome, but I don't see this immediately available with the Slack orb